src/Core/src/Credentials/PodIdentityProvider.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Core\Credentials;
+
+use AsyncAws\Core\Configuration;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Provides Credentials for running Pod Identity inside an EKS Cluster.
+ *
+ * @see https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html
+ */
+final class PodIdentityProvider implements CredentialProvider
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var HttpClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * @var TokenFileLoader
+     */
+    private $tokenFileLoader;
+
+    public function __construct(?HttpClientInterface $httpClient = null, ?LoggerInterface $logger = null, ?TokenFileLoader $tokenFileLoader = null)
+    {
+        $this->logger = $logger ?? new NullLogger();
+        $this->httpClient = $httpClient ?? HttpClient::create();
+        $this->tokenFileLoader = $tokenFileLoader ?? new TokenFileLoader();
+    }
+
+    public function getCredentials(Configuration $configuration): ?Credentials
+    {
+        $fullUri = $configuration->get(Configuration::OPTION_POD_IDENTITY_CREDENTIALS_FULL_URI);
+        $tokenFile = $configuration->get(Configuration::OPTION_POD_IDENTITY_AUTHORIZATION_TOKEN_FILE);
+
+        // introduces an early exit if the env variable is not set.
+        if (empty($fullUri) || empty($tokenFile)) {
+            return null;
+        }
+
+        // fetch token from file
+        try {
+            $tokenFileContent = $this->tokenFileLoader->getTokenFileContent($tokenFile);
+        } catch (\Exception $e) {
+            $this->logger->warning('"Error reading PodIdentityTokenFile "{tokenFile}.', ['tokenFile' => $tokenFile, 'exception' => $e]);
+
+            return null;
+        }
+
+        // fetch credentials from pod identity agent endpoint
+        try {
+            $response = $this->httpClient->request('GET', $fullUri, ['headers' => ['Authorization' => $tokenFileContent]]);
```

Comment:
the official AWS SDK checks if the URL is valid, I believe we should do the same:

https://github.com/aws/aws-sdk-php/blob/7412a44da62fd607efbaac4084e69d6621f29de1/src/Credentials/EcsCredentialProvider.php#L231-L249